core/src/main/java/com/datastax/oss/driver/internal/core/metadata/DefaultTopologyMonitor.java

```diff
@@ -133,10 +156,29 @@ public DefaultTopologyMonitor(InternalDriverContext context) {
     savePort(channel);
 
     CompletionStage<AdminResult> localQuery = query(channel, "SELECT * FROM system.local");
-    CompletionStage<AdminResult> peersQuery = query(channel, "SELECT * FROM system.peers");
+    CompletionStage<AdminResult> peersV2Query = query(channel, "SELECT * FROM system.peers_v2");
+    CompletableFuture<AdminResult> peersQ = new CompletableFuture<>();
+
+    peersV2Query.whenComplete(
+        (r, t) -> {
+          if (t != null) {
+            // The query to system.peers_v2 failed, whe should not attempt this query in the future.
+            // Since this method should be run as part of the control connection initialization, we
+            // shouldn't
+            // need this fallback anywhere else.
+            this.isSchemaV2 = false;
+            query(channel, "SELECT * FROM system.peers")
+                .whenComplete(
+                    (r2, t2) -> {
+                      peersQ.complete(r2);
+                    });
+          } else {
+            peersQ.complete(r);
+          }
+        });
```

Comment:
Could you chain `.exceptionally(UncaughtExceptions::log)` here?
It's very unlikely that an exception would be thrown from your `whenComplete` callback, but if that happens it would be silently ignored with the current code (because we discard the resulting future). With the log we at least detect it.